packages/appium/docs/en/ecosystem/build-drivers.md

```diff
@@ -2,4 +2,617 @@
+helper file), you can always construct your own too:
+
+```js
+import {logging} from 'appium/support';
+const log = logging.getLogger('MyDriver');
+```
+
+## Further possibilities for Appium drivers
+
+These are things your driver *can* do to take advantage of extra driver features or do its job more
+conveniently.
+
+### Add a schema for custom command line arguments
+
+You can add custom CLI args if you want your driver to receive data from the command line when the
+Appium server is started (for example, ports that a server administrator should set that should not
+be passed in as capabilities.
+
+To do this you define a `schema` field within the `appium` field in `package.json`, pointing to
+a JS or JSON file defining an argument schema. (See the schema in `fake-driver`, for example).
+Arguments you define there will be available as a `--driver-<name>-<arg-name>` flag on the command
+line. If you define `my-arg` in the schema, for example, and if your driver's name is `awesome`,
+then users will be able to set your arg via `--driver-awesome-my-arg`.
+
+### Add driver scripts
+
+Sometimes you might want users of your driver to be able to run scripts outside the context of
+a session (for example, to run a script that pre-builds aspects of your driver). To support this,
+you can add a map of script names and JS files to the `scripts` field within your Appium manifest.
+So let's say you've created a script in your project that lives in a `scripts` directory in your
+project, named `driver-prebuild.js`. Then you could add a `scripts` field like this:
+
+```json
+{
+    "scripts": {
+        "prebuild": "./scripts/driver-prebuild.js"
+    }
+}
+```
+
+Now, assuming your driver is named `mydriver`, users of your driver can run `appium driver run
+mydriver prebuild`, and your script will execute.
+
+### Proxy commands to another WebDriver implementation
+
+A very common design architecture for Appium drivers is to have some kind of platform-specific
+WebDriver implementation that the Appium driver interfaces with. For example, the Appium
+UiAutomator2 driver interfaces with a special (Java-based) server running on the Android device. In
+webview mode, it also interfaces with Chromedriver.
+
+If you find yourself in this situation, it is extremely easy to tell Appium that your driver is
+just going to be proxying WebDriver commands straight to another endpoint.
+
+First, let Appium know that your driver *can* proxy by implementing the `canProxy` method:
+
+```
+canProxy() {
+    return true;
+}
+```
+
+Next, tell Appium which WebDriver routes it should *not* attempt to proxy (there often end up being
+certain routes that you don't want to forward on):
+
+```
+getProxyAvoidList() {
+    return [
+        ['POST', new RegExp('^/session/[^/]+/appium')]
+    ];
+}
+```
+
+The proxy avoidance list should be an array of arrays, where each inner array has an HTTP method as
```

Comment:
seems like this ought to support a different data structure someday

Reply:
👍